lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -11939,8 +11939,7 @@ using namespace Js;
 
         Assert(IsInlineSegment(src, instance) == IsInlineSegment(dst, static_cast<T*>(this)));
 
-        AssertOrFailFast(dst->size <= src->size);
-        CopyArray(dst->elements, dst->size, src->elements, dst->size);
+        CopyArray(dst->elements, dst->size, src->elements, min(dst->size, src->size));
```

Comment:
It would be nice if we could make this a little bit more self-explanatory for future maintainers. Would something like the following be equivalent?

```cpp
uint32 sourceSize = source->size;

if (IsInlineSegment(src, instance))
{
    // ...code...

    // Comment here explaining things
    sourceSize = min(sourceSize, inlineChunkSize);
    dst->size = sourceSize;
}
else
{
    // ...code...
}

// ...code...

CopyArray(dst->elements, dst->size, src->elements, sourceSize);
```

Reply:
will make a change